Configuration/Settings.yaml

```diff
@@ -29,8 +42,22 @@ Flowpack:
         # ignored arguments are not part of the cache identifier but do not
         # prevent caching either. Use this for arguments that are meaningless for
         # the backend like utm_campaign
+        # Example:
+        #  - 'utm_campaign'
+        #  => If the "utm_campaign" argument is present in the request, it will be cached.
+        #     But if a "utm_source" argument is present, the request will not be cached.
+        # Important: Only evaluated if "respect" is empty!
         ignore: []
 
+        # respected arguments are relevant for the backend and will skip the caching.
+        # All other arguments will be ignored and cached as if they weren't there.
+        # Example:
+        #  - 'search'
+        #  => If the "search" argument is present in the request, it will not be cached.
+        #     But if a "utm_source" argument is present, the request will be cached as if the argument wouldn't be there.
+        # Important: Only evaluated if "ignore" is empty!
+        respect: []
```

Comment:
Setting empty Arrays is a bit nasty, as this will override already set configurations in packages loaded before 🤔

Reply:
Why should another package write this config before this one?

Reply:
Happens all the time when dependencies between packages are not properly declared. Not actually declaring this key but only documenting it helps in those cases.

Reply:
> But it's not the business of a package to deal with incorrect dependencies.

We have a development package for example, we load in require-dev.
Loading a package there, will reset settings already set in another package. 

So i personally prefer to not have empty configurations in configs @Sebobo

Reply:
> [paavo](https://github.com/paavo) reviewed [yesterday](https://github.com/Flowpack/Flowpack.FullPageCache/pull/22#pullrequestreview-2641352114)

Oh, i just wanted to drop a comment 🙈 😆 
Nothing reviewed so far 😆

Reply:
@paavo but if that package has a "suggest" or "require" on this package, then then merge order is also fine and you won't have problems.

Or is the actual problem using arrays instead of objects?

Reply:
Let me make an example @Sebobo with [Sitegeist.MagicWand](https://github.com/sitegeist/Sitegeist.MagicWand):
We have our Site-Package where we define MagicWand Settings `clonePresets:  ...`.

Then we have a development package we load in require-dev (to only load this on DEV Environments).
This development package requires MagicWand.
And MagicWand sets default `clonePresets: []` and overrides the Configuration in the Site-Package.

> Or is the actual problem using arrays instead of objects?
The Problem it, that configurations get overridden. 

---

Yes, FullpageCache is probably the opposite of a Dev-Package 😆  
But i really prefer not overriding configurations with empty values.

Reply:
Then your site package needs also an `require-dev` or `suggest` entry for your dev package.

Reply:
I am undecided wether we should help here. On the one hand it is an easy fix to not actually write the empty config array. On the other hand projects with basically random loading order between interdepending packages will run into issues eventually that are hard to debug. So we may actually do those a favor ...